astropy/config/tests/test_configs.py

```diff
@@ -38,18 +38,118 @@ def test_paths():
 
 
 @pytest.mark.parametrize(
-    "environment_variable,func",
+    "env_var_template",
     [
-        # Regression test for #17514 - XDG_CACHE_HOME had no effect
-        pytest.param("XDG_CACHE_HOME", paths.get_cache_dir_path, id="cache"),
-        pytest.param("XDG_CONFIG_HOME", paths.get_config_dir_path, id="config"),
+        pytest.param("XDG_{}_HOME", id="xdg"),
+        pytest.param("ASTROPY_{}_DIR", id="astropy"),
     ],
 )
-def test_xdg_variables(monkeypatch, tmp_path, environment_variable, func):
-    config_dir = tmp_path / "astropy"
-    config_dir.mkdir()
+@pytest.mark.parametrize("dir_type", ["CACHE", "CONFIG"])
+def test_env_variables(monkeypatch, tmp_path, env_var_template, dir_type):
+    # ignore global state of the test session
+    monkeypatch.setenv(f"ASTROPY_{dir_type}_DIR", "foo")
+    monkeypatch.delenv(f"ASTROPY_{dir_type}_DIR")
+    monkeypatch.setenv(f"XDG_{dir_type}_HOME", "foo")
+    monkeypatch.delenv(f"XDG_{dir_type}_HOME")
+
+    match dir_type:
+        case "CACHE":
+            cls = paths.set_temp_cache
+            func = paths.get_cache_dir_path
```

Comment:
`func` should be a part of the parametrization.

Reply:
For `func` this should be self-evident. For `cls` my understanding is that we only need manage it because the underlying code is not thread safe, so I hoped that making it thread safe makes `cls` unnecessary.